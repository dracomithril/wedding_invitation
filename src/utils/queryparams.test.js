const { stringify } = require('./queryparams');

describe('queryParams', () => {
	it('should be able to obtain string from object', () => {
		const query = stringify({ data: 'z' });
		expect(query).toEqual('data=z');
	});

	it('should be able to add stringify undefined values', () => {
		const query = stringify({ data: undefined });
		expect(query).toEqual('data');
	});

	it('should be able to parse boolean values', () => {
		const query = stringify({ data: true });
		expect(query).toEqual('data=true');
	});

	it('should encode 0 correctly', () => {
		const query = stringify({ data: 0 });
		expect(query).toEqual('data=0');
	});

	it('should be able to url encode string', () => {
		const query = stringify({ data: 'Wesele GrezelkÃ³w' });
		expect(query).toMatchSnapshot();
	});

	it('should be able to connect multiple keys', () => {
		const query = stringify({ data: 'space', module: undefined, data2: 'zamarin' });
		expect(query).toMatchSnapshot();
	});

	it('should create full query params', () => {
		const queries = {
			dates: '20200501T120000Z/20200502T040000Z',
			text: 'Space suits warp with life!',
			location: 'Reproduce patiently like a bare crew.',
			details: undefined,
			sf: true,
			ctz: 'America/New_York'
		};
		const query = stringify(queries);
		expect(query).toMatchSnapshot();

	});
});