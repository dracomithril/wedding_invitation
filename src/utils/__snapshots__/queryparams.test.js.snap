// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`queryParams should be able to connect multiple keys 1`] = `"data=space&module&data2=zamarin"`;

exports[`queryParams should be able to url encode string 1`] = `"data=Wesele+Grezelk√≥w"`;

exports[`queryParams should create full query params 1`] = `"dates=20200501T120000Z/20200502T040000Z&text=Space+suits+warp+with+life!&location=Reproduce+patiently+like+a+bare+crew.&details&sf=true&ctz=America/New_York"`;
